arangod/Aql/AqlValue.h

```diff
@@ -272,6 +275,79 @@ struct AqlValue final {
     } longNumberMeta;
     static_assert(sizeof(longNumberMeta) == 16,
                   "VPACK_INLINE_INT64 layout is not 16 bytes!");
+
+    // VPACK_SUPERVISED_SLICE
+    struct {
+      velocypack::Slice toSlice() const noexcept {
+        return velocypack::Slice(reinterpret_cast<uint8_t const*>(
+            pointer + sizeof(arangodb::ResourceMonitor*)));
+      }
+      uint64_t getLength() const noexcept {
+        return velocypack::Slice(pointer + sizeof(arangodb::ResourceMonitor*))
+            .byteSize();
+      }
+
+      uint64_t getOrigin() const noexcept {
+        if constexpr (basics::isLittleEndian()) {
+          return (lengthOrigin & 0x000000000000ff00ULL) >> 8;
+        } else {
+          return (lengthOrigin & 0x00ff000000000000ULL) >> 48;
+        }
```

Comment:
This is weird, but since we also have this for the managed slice I suppose we can keep it for now and clean this up separately.